### PR TITLE
added search field not case sensitive

### PR DESCRIPTION
**Pull request by danielnottingham**
Fix issue: #10 
find by name

At home page, type some string and show only itens that have that string. Example, edimo, Edimilson and Daniel. If searh for ed shoud return edimo and Edmilson. The search is not case sensitive

Test coverage:
image

link Heroku app: https://fierce-springs-99712.herokuapp.com/

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,5 +1,10 @@
 class User < ApplicationRecord
-    validates :name, :phone, presence: true
-    validates :name, format: { with: /\A[a-zA-Z]+\z/}
-    validates :phone, numericality: { only_integer: true }
+  validates :name, :phone, presence: true
+  validates :name, format: { with: /\A[a-zA-Z]+\z/}
+  validates :phone, numericality: { only_integer: true }
+
+  def self.search(query)
```
Comment (edimossilva) [inline app/models/user.rb:6]: Use scope instead of this method

**Review by edimossilva — CHANGES_REQUESTED** [review]
use scope instead of static method

**Review by edimossilva — CHANGES_REQUESTED** [review]
tests?

File: app/models/user.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 class User < ApplicationRecord
-    validates :name, :phone, presence: true
-    validates :name, format: { with: /\A[a-zA-Z]+\z/}
-    validates :phone, numericality: { only_integer: true }
+  validates :name, :phone, presence: true
+  validates :name, format: { with: /\A[a-zA-Z]+\z/ }
+  validates :phone, numericality: { only_integer: true }
+  scope :search, ->(query) { where('name ilike ?', "#{query}%") }
```
Comment (edimossilva) [inline app/models/user.rb:5]: rename this to `search_by_name`

File: spec/controllers/users_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      #assert
+      assigns(:users).should eq([user1, user2])
+    end
+
+    it 'finds by name with da' do
+      #action
+      get :index, params: { search: 'Da' }
+      #assert
+      assigns(:users).should eq([user])
+    end
```
Comment (edimossilva) [inline spec/controllers/users_controller_spec.rb:22]: we need the other test scenarios:
1) when does not find any user
2) when search is empty

**Review by edimossilva — CHANGES_REQUESTED** [review]
1) enforce rubocop rules
2) tag the issue into PR description
3) post test coverage report
4) put link to heroku deploy on description

File: spec/controllers/users_controller_spec.rb (hunk 3 above)
Comment (danielnottingham) [reply spec/controllers/users_controller_spec.rb:22]: I still can't post the test covered, only when I'm going to pass the request tests to the controller

File: spec/controllers/users_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    it 'finds by name with da' do
+      #action
+      get :index, params: { search_by_name: 'Da' }
+      #assert
+      assigns(:users).should eq([user])
+    end
+
+    it 'when does not find any user' do
+      #action
```
Comment (edimossilva) [inline spec/controllers/users_controller_spec.rb:25]: it is good to create users here even if you know that they are not coming. Think like this, if some issue happens and the method returns all users when you use a query that supposed to return an empty array, this test will not catch it

File: spec/controllers/users_controller_spec.rb (hunk 3 above)
Comment (danielnottingham) [reply spec/controllers/users_controller_spec.rb:22]: Screenshot from 2020-09-02 11-02-02

Comment (edimossilva) [reply spec/controllers/users_controller_spec.rb:22]: should be at the pr description

File: spec/controllers/users_controller_spec.rb (hunk 4 above)
Comment (edimossilva) [reply spec/controllers/users_controller_spec.rb:25]: it is already done

**Review by edimossilva — APPROVED** [review]

**Pull request merged**